service/matching/task_queue_partition_manager.go

```diff
@@ -686,7 +686,11 @@ func (pm *taskQueuePartitionManagerImpl) Describe(
 			if !found {
 				// still add it as a v2 version because user explicitly asked for the stats, we'd
 				// make sure to load the TQ.
-				versions[PhysicalTaskQueueVersion{buildId: b}] = true
+				dv := worker_versioning.ExternalWorkerDeploymentVersionFromStringV31(b)
```

Comment:
Is this the right helper to call? There are too many :P

Furthermore, do we need to guard against a parsing issue here? I suspect it happens earlier. Will have to check ...